File: algorithms/0867.transpose-matrix/solution.rs
impl Solution {
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut res: Vec<Vec<i32>> = vec![vec![0; matrix.len()]; matrix[0].len()];
        for i in 0..matrix.len() {
            for j in 0..matrix[0].len() {
                res[j][i] = matrix[i][j]
            }
        }
        res
    }
}

File: algorithms/0867.transpose-matrix/solution.go
func transpose(matrix [][]int) [][]int {
	rows := len(matrix)
	cols := len(matrix[0])

	res := make([][]int, cols)
	for row := 0; row < cols; row++ {
		res[row] = make([]int, rows)
		for col := 0; col < rows; col++ {
			res[row][col] = matrix[col][row]
		}
	}

	return res
}
